admiral/debug_functions.sqf

```diff
@@ -287,6 +287,88 @@ adm_debug_fnc_updateLineMarker = {
     DEBUG("admiral.debug",FMT_3("Updated line marker '%1' position between '%2' and '%3'.",_markerName,_posFrom,_posTo));
 };
 
+adm_debug_fnc_createDebugCounterMarker = {
+    FUN_ARGS_3(_side,_type,_xPos);
+    
+    PVT_2(_marker,_markerType);
+    _markerType = if (_type != "total") then {
+        _type;
+    } else {
+        COUNTER_DEBUG_MARKER_TOTAL;
+    };
+    _marker = [format ["adm_counter_%1_%2", _side,_type], [_xPos, 50, 0], "ICON", _markerType, [_side] call adm_debug_fnc_getSideColor, COUNTER_DEBUG_MARKER_SIZE] call adm_common_fnc_createLocalMarker;
+    _marker setMarkerTextLocal "0";
+    
+    _marker;
+};
+
+adm_debug_fnc_createDebugFactionCounterMarkers = {
+    FUN_ARGS_2(_side,_xPos);
+
+    DECLARE(_debugMarkers) = [];
+    PVT_1(_marker);
+    {
+        _marker = [_side, _x, _xPos] call adm_debug_fnc_createDebugCounterMarker;
+        _debugMarkers pushBack _marker;
+        _xPos = _xPos + COUNTER_DEBUG_MARKER_X_INCREMENT;
+    } foreach GROUP_TYPE_DEBUG_MARKERS;
+    _marker = [_side, "total", _xPos] call adm_debug_fnc_createDebugCounterMarker;
+    _debugMarkers pushBack _marker;
+    _xPos = _xPos + COUNTER_DEBUG_MARKER_X_INCREMENT;
+    DEBUG("admiral.debug",FMT_2("Created counter Markers '%1' for side '%2'.",_debugMarkers,_side));
+};
+
+adm_debug_fnc_createAllDebugCounterMarkers = {
+    DECLARE(_xPos) = COUNTER_DEBUG_MARKER_X_POS;
+    {
+        [_x,_xPos] call adm_debug_fnc_createDebugFactionCounterMarkers;
+        _xPos = _xPos + (4 * COUNTER_DEBUG_MARKER_X_INCREMENT);
+    } foreach SIDE_ARRAY;
+};
+
+adm_debug_fnc_returnUnitCount = {
```

Comment:
This function is not needed. The only difference in the calculation on return values here are the groups for the group types. Also `total` is calculated differently from the rest. Since you are already calling this with `total` from `336` explicitly, you can just put the calculation call there.

As for the group types, instead of doing a `switch` on them, you should create an array with the groups needed for each group type in the correct order. Then you can select the correct array containing the groups by `_forEachIndex`. If this doesn't make any sense just poke me and I'll show you what I meant.